compute_endpoint/globus_compute_endpoint/executors/high_throughput/interchange.py

```diff
@@ -565,21 +574,20 @@ def _status_report_loop(self, kill_event, status_report_queue: queue.Queue):
                 len(task_status_deltas),
             )
 
+            # For multi-chunk reports ("lots-o-tasks"), the state won't change *that*
+            # much each iteration, so cache the result
+            global_state = self.get_global_state_for_status_report()
+
             # The result processor will gracefully handle any size message, but
             # courtesy says to chunk work; 4,096 is empirically chosen to be plenty
             # "bulk enough," but not rude.
             for tsd_chunk in chunk_by(task_status_deltas.items(), 4_096):
-                try:
-                    msg = EPStatusReport(
-                        self.endpoint_id,
-                        self.get_global_state_for_status_report(),
-                        dict(tsd_chunk),
-                    )
-                    status_report_queue.put(msg.pack())
-                except Exception:
-                    log.exception("Unable to create or send EP status report.")
-                    log.debug("Attempted to send chunk: %s", tsd_chunk)
-                    # ignoring so that the thread continues; "it's just a status"
+                _enqueue_status_report(global_state, dict(tsd_chunk))
```

Comment:
I would have found doing the if check first makes it clearer, but it doesn't make too much difference.
```
if task_status_deltas:
    for tsd_chunk in chunk_by(task_status_deltas.items(), 4_096):
           _enqueue_status_report(global_state, dict(tsd_chunk))
else:
    _enqueue_status_report(global_state, {})
```

Reply:
Fair.  "One less indent level."  🤷